pkg/trace/api/evp_intake_proxy.go

```diff
@@ -0,0 +1,232 @@
+	if apiKey == "" {
+		apiKey = conf.APIKey()
+	}
+	endpoint := conf.EVPProxy.DDURL
+	if endpoint == "" {
+		endpoint = conf.Site
+	}
+	mainEndpoint := config.Endpoint{Host: endpoint, APIKey: apiKey}
+	endpoints := []config.Endpoint{mainEndpoint}
+	for host, keys := range conf.EVPProxy.AdditionalEndpoints {
+		for _, key := range keys {
+			endpoints = append(endpoints, config.Endpoint{
+				Host:   host,
+				APIKey: key,
+			})
+		}
+	}
+	return endpoints
+}
+
+// evpProxyHandler returns an HTTP handler for the /evp_proxy API.
+// Depending on the config, this is a proxying handler or a noop handler.
+func (r *HTTPReceiver) evpProxyHandler() http.Handler {
+	// r.conf is populated by cmd/trace-agent/config/config.go
+	if !r.conf.EVPProxy.Enabled {
+		return evpProxyErrorHandler("Has been disabled in config")
+	}
+	endpoints := evpProxyEndpointsFromConfig(r.conf)
+	transport := r.conf.NewHTTPTransport()
+	logger := stdlog.New(log.NewThrottled(5, 10*time.Second), "EVPProxy: ", 0) // limit to 5 messages every 10 seconds
+	handler := evpProxyForwarder(r.conf, endpoints, transport, logger)
+	return http.StripPrefix("/evp_proxy/v1", handler)
+}
+
+// evpProxyErrorHandler returns an HTTP handler that will always return
+// http.StatusMethodNotAllowed along with a clarification.
+func evpProxyErrorHandler(message string) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		msg := fmt.Sprintf("EVPProxy is disabled: %v", message)
+		http.Error(w, msg, http.StatusMethodNotAllowed)
+	})
+}
+
+// evpProxyForwarder creates an http.ReverseProxy which can forward payloads to
+// one or more endpoints, based on the request received and the Agent configuration.
+// Headers are not proxied, instead we add our own known set of headers.
+// See also evpProxyTransport below.
+func evpProxyForwarder(conf *config.AgentConfig, endpoints []config.Endpoint, transport http.RoundTripper, logger *stdlog.Logger) http.Handler {
+	return &httputil.ReverseProxy{
+		Director: func(req *http.Request) {
+			req.Header["X-Forwarded-For"] = nil // Prevent setting X-Forwarded-For
+		},
+		ErrorLog:  logger,
+		Transport: &evpProxyTransport{transport, endpoints, conf},
+	}
+}
+
+// evpProxyTransport sends HTTPS requests to multiple targets using an
+// underlying http.RoundTripper. API keys are set separately for each target.
+// When multiple endpoints are in use the response from the first endpoint
+// is proxied back to the client, while for all aditional endpoints the
+// response is discarded.
+type evpProxyTransport struct {
+	transport http.RoundTripper
+	endpoints []config.Endpoint
+	conf      *config.AgentConfig
+}
+
+func (t *evpProxyTransport) RoundTrip(req *http.Request) (rresp *http.Response, rerr error) {
+	if req.Body != nil && t.conf.EVPProxy.MaxPayloadSize > 0 {
+		req.Body = apiutil.NewLimitedReader(req.Body, t.conf.EVPProxy.MaxPayloadSize)
+	}
+
+	beginTime := time.Now()
+	metricTags := []string{}
```

Comment:
In other places in the repo we've been calling these `now` (or `start`) and `tags`. Do you mind changing that so we can be consistent? If you've seen it done otherwise in some part of the code, then sorry about that. Do let me know.